sweetpea/_internal/primitive.py

```diff
@@ -674,6 +685,28 @@ def desugar_for_weights(self, replacements: dict):
             f = DerivedFactor(self.name, [replacements[l] for l in self.levels])
             replacements[self] = [f, f]
 
+    def test_trial(self, i, trial_sequence):
+        res = True
+        for level in self.levels:
+            if trial_sequence[self.name][i] == level.name:
+                args = []
+                for f in level.window.factors:
+                    levels = trial_sequence[f.name]
+                    for j in range(level.window.width):
+                        idx = i - (level.window.width - 1) + j
+                        if idx >= 0:
+                            args.append(levels[idx])
+                        else:
+                            args.append(None)
+                if level.window.width > 1:
+                    args = list(chunk_dict(args, level.window.width))
```

Comment:
The code above looks like `DerivedLevel._trial_arguments` (although it looks like the docstring there is out-of-date). Does `args = level._trial_arguments(trial_sequence, i)` work here?

Reply:
Yes, i copied it from there. Thera are two differences (I don't know if those are errors in _trial_arguments so I didn't want to touch an existing function):
in the _trial_arguments, instead of f.name it is just f (line 604)
in the _trial_arguments, instead of levels[idx] it is levels[idx].name (line 698)

Reply:
Ah, that difference makes sense, because `_trial_arguments` was created for an intermediate representation with factor and level objects instead of names. It would make sense for `_trial_arguments` to take another argument to say which format the sample is in, though, instead of copying the code.

Reply:
I added a argument with the type string and the default value "object". The value can either be "object" or "name". Does that make sense?